crates/virtual-dom-rs/src/patch.rs

```diff
@@ -140,40 +152,32 @@ fn find_nodes(
     for i in 0..child_node_count {
         let node = children.item(i).unwrap();
 
-        let element = node.dyn_into::<Element>();
-
-        if element.is_ok() {
-            find_nodes(
-                element.ok().unwrap(),
-                cur_node_idx,
-                nodes_to_find,
-                nodes_to_patch,
-                text_nodes_to_patch,
-            );
-        } else {
-            let text_or_comment_node = element.err().unwrap();
-
-            // At this time we do not support user entered comment nodes, so if we see a comment
-            // then it was a delimiter created by virtual-dom-rs in order to ensure that two
-            // neighboring text nodes did not get merged into one by the browser. So we skip
-            // over this virtual-dom-rs generated comment node.
-            if text_or_comment_node.node_type() == Node::COMMENT_NODE {
-                continue;
+        match node.node_type() {
+            Node::ELEMENT_NODE => {
+                find_nodes(node, cur_node_idx, nodes_to_find, element_nodes_to_patch, text_nodes_to_patch);
             }
+            Node::TEXT_NODE => {
+                if nodes_to_find.get(&cur_node_idx).is_some() {
+                    text_nodes_to_patch.insert(*cur_node_idx, node.unchecked_into());
+                }
 
-            if nodes_to_find.get(&cur_node_idx).is_some() {
-                let text_node = text_or_comment_node;
-                text_nodes_to_patch.insert(*cur_node_idx, text_node);
+                *cur_node_idx += 1;
+            }
+            Node::COMMENT_NODE => {
+                // At this time we do not support user entered comment nodes, so if we see a comment
+                // then it was a delimiter created by virtual-dom-rs in order to ensure that two
+                // neighboring text nodes did not get merged into one by the browser. So we skip
+                // over this virtual-dom-rs generated comment node.
+            }
+            _other => {
+                // Ignoring unsupported child node type
```

Comment:
One thing that comes to mind - Is this block even reachable in any way right now?

Since VirtualNode doesn't define them I'm wondering if it's even possible to get to this block in the first place?

Reply:
Yes, because the `Node` (note: Not a `VirtualNode`) that is passed into the function may have an arbitrary type.